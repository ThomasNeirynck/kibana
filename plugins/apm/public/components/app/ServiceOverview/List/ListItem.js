import React from 'react';
import styled from 'styled-components';
import { RelativeLink } from '../../../../utils/url';
import { KuiTableRow, KuiTableRowCell } from 'ui_framework/components';
import { px, unit, fontSizes } from '../../../../style/variables';
import { RIGHT_ALIGNMENT } from 'ui_framework/services';
import { asMillisWithDefault } from '../../../../utils/formatters';
import numeral from '@elastic/numeral';

const ServiceNameCell = styled(KuiTableRowCell)`
  max-width: ${px(unit * 2)};
`;

const AppLink = styled(RelativeLink)`
  font-size: ${fontSizes.large};
`;

function formatString(value) {
  return value || 'N/A';
}

function formatNumber(value) {
  const formatted = numeral(value).format('0.0');
  return formatted <= 0.1 ? '< 0.1' : formatted;
}

function ListItem({ service }) {
  const {
    serviceName,
    agentName,
    transactionsPerMinute,
    errorsPerMinute,
    avgResponseTime
  } = service;

  return (
    <KuiTableRow>
      <ServiceNameCell>
        <AppLink path={`${serviceName}/transactions`}>
          {formatString(serviceName)}
        </AppLink>
      </ServiceNameCell>
      <KuiTableRowCell>{formatString(agentName)}</KuiTableRowCell>
      <KuiTableRowCell align={RIGHT_ALIGNMENT}>
        {asMillisWithDefault(avgResponseTime)}
      </KuiTableRowCell>
      <KuiTableRowCell align={RIGHT_ALIGNMENT}>
        {formatNumber(transactionsPerMinute)}
      </KuiTableRowCell>
      <KuiTableRowCell align={RIGHT_ALIGNMENT}>
        {formatNumber(errorsPerMinute)}
      </KuiTableRowCell>
    </KuiTableRow>
  );
}

export default ListItem;
