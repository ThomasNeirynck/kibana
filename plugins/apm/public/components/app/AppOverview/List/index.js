import React, { Component } from 'react';

import APMTable, {
  AlignmentKuiTableHeaderCell
} from '../../../shared/APMTable';

import ListItem from './ListItem';

class List extends Component {
  render() {
    const { items, changeAppSorting, appSorting } = this.props;

    const renderHead = () => {
      const cells = [
        { key: 'appName', label: 'Name' },
        { key: 'agentName', label: 'Agent' },
        {
          key: 'avgResponseTime',
          label: 'Avg. response time',
          alignRight: true
        },
        {
          key: 'transactionsPerMinute',
          label: 'Transactions / min.',
          alignRight: true
        },
        { key: 'errorsPerMinute', label: 'Errors  / min.', alignRight: true }
      ].map(({ key, label, alignRight }) => (
        <AlignmentKuiTableHeaderCell
          key={key}
          onSort={() => changeAppSorting(key)}
          isSorted={appSorting.key === key}
          isSortAscending={!appSorting.descending}
          className={alignRight ? 'kuiTableHeaderCell--alignRight' : ''}
        >
          {label}
        </AlignmentKuiTableHeaderCell>
      ));

      return cells;
    };

    const renderBody = apps => {
      return apps.map(app => {
        return <ListItem key={app.appName} app={app} />;
      });
    };

    return (
      <APMTable
        resultsLimit={500}
        searchableFields={['appName', 'agentName']}
        items={items}
        emptyMessageHeading="No apps with data in the selected time range."
        renderHead={renderHead}
        renderBody={renderBody}
      />
    );
  }
}

export default List;
