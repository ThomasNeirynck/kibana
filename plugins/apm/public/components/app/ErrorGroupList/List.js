import React from 'react';
import Loader from './Loader';
import ListItem from './ListItem';
import styled from 'styled-components';
import { units, px, colors, borderRadius } from '../../../style/variables';

const ErrorsContainer = styled.div`
  position: relative;
  overflow: hidden;
  padding: 0;
  border: 1px solid ${colors.elementBorder};
  border-radius: ${borderRadius};
`;

const Table = styled.table`width: 100%;`;

const TableHeading = styled.th`
  text-align: right;
  border-bottom: 1px solid ${colors.elementBorder};
  border-left: 1px solid ${colors.tableBorder};
  padding: ${px(units.minus)};
  position: relative;
  cursor: pointer;

  &:first-child {
    border-left: 0;
    text-align: left;
  }
`;

function List({ appName, list }) {
  return (
    <ErrorsContainer>
      <Table>
        <thead>
          <tr>
            <TableHeading>Error grouping</TableHeading>
            <TableHeading>Occurrences</TableHeading>
          </tr>
        </thead>

        <tbody>
          <Loader status={list.status} />

          {list.data.map(error => {
            return (
              <ListItem
                key={error.groupingId}
                appName={appName}
                error={error}
              />
            );
          })}
        </tbody>
      </Table>
    </ErrorsContainer>
  );
}

export default List;
