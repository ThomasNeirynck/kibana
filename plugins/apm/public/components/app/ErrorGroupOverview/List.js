import React from 'react';
import styled from 'styled-components';
import { Table, TableHead, TableLoader } from '../../shared/Table';
import ListItem from './ListItem';
import { colors, borderRadius } from '../../../style/variables';

const ErrorsContainer = styled.div`
  position: relative;
  overflow: hidden;
  padding: 0;
  border: 1px solid ${colors.elementBorder};
  border-radius: ${borderRadius};
`;

function List({ appName, list }) {
  return (
    <ErrorsContainer>
      <Table>
        <thead>
          <tr>
            <TableHead>Error grouping</TableHead>
            <TableHead>Occurrences</TableHead>
          </tr>
        </thead>

        <tbody>
          <TableLoader status={list.status} columns={2} />

          {list.data.map(error => {
            return (
              <ListItem
                key={error.groupingId}
                appName={appName}
                error={error}
              />
            );
          })}
        </tbody>
      </Table>
    </ErrorsContainer>
  );
}

export default List;
