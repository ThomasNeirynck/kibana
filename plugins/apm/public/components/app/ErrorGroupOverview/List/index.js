import React, { Component } from 'react';

import APMTable, {
  AlignmentKuiTableHeaderCell
} from '../../../shared/APMTable';
import ListItem from './ListItem';

class List extends Component {
  render() {
    const {
      appName,
      items,
      changeErrorGroupSorting,
      errorGroupSorting
    } = this.props;

    const renderHead = () => {
      const cells = [
        { key: 'groupId', label: 'Group ID' },
        { key: 'message', label: 'Error message and culprit' },
        {
          key: 'occurrenceCount',
          label: 'Group occurrences',
          alignRight: true
        },
        {
          key: 'latestOccurrenceAt',
          label: 'Latest occurrence',
          alignRight: true
        }
      ].map(({ key, label, alignRight }) => (
        <AlignmentKuiTableHeaderCell
          key={key}
          className={alignRight ? 'kuiTableHeaderCell--alignRight' : ''}
          onSort={() => changeErrorGroupSorting(key)}
          isSorted={errorGroupSorting.key === key}
          isSortAscending={!errorGroupSorting.descending}
        >
          {label}
        </AlignmentKuiTableHeaderCell>
      ));

      return cells;
    };

    const renderBody = errorGroups => {
      return errorGroups.map(error => {
        return <ListItem key={error.groupId} appName={appName} error={error} />;
      });
    };

    return (
      <APMTable
        searchableFields={['groupId', 'culprit', 'message']}
        items={items}
        emptyMessageHeading="No errors in the selected time range."
        resultsLimitOrder="latest occurrence in group"
        renderHead={renderHead}
        renderBody={renderBody}
      />
    );
  }
}

export default List;
