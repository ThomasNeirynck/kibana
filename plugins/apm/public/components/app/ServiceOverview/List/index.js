import React, { Component } from 'react';

import APMTable, {
  AlignmentKuiTableHeaderCell
} from '../../../shared/APMTable';

import ListItem from './ListItem';

class List extends Component {
  render() {
    const { items, changeServiceSorting, serviceSorting } = this.props;

    const renderHead = () => {
      const cells = [
        { key: 'serviceName', label: 'Name' },
        { key: 'agentName', label: 'Agent' },
        {
          key: 'avgResponseTime',
          label: 'Avg. response time',
          alignRight: true
        },
        {
          key: 'transactionsPerMinute',
          label: 'Transactions / min.',
          alignRight: true
        },
        { key: 'errorsPerMinute', label: 'Errors  / min.', alignRight: true }
      ].map(({ key, label, alignRight }) => (
        <AlignmentKuiTableHeaderCell
          key={key}
          onSort={() => changeServiceSorting(key)}
          isSorted={serviceSorting.key === key}
          isSortAscending={!serviceSorting.descending}
          className={alignRight ? 'kuiTableHeaderCell--alignRight' : ''}
        >
          {label}
        </AlignmentKuiTableHeaderCell>
      ));

      return cells;
    };

    const renderBody = services => {
      return services.map(service => {
        return <ListItem key={service.serviceName} service={service} />;
      });
    };

    return (
      <APMTable
        resultsLimit={500}
        searchableFields={['serviceName', 'agentName']}
        items={items}
        emptyMessageHeading="No services with data in the selected time range."
        renderHead={renderHead}
        renderBody={renderBody}
      />
    );
  }
}

export default List;
