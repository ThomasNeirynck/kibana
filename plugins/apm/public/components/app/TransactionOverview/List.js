import React from 'react';
import styled from 'styled-components';
import { Table, TableHead, TableLoader } from '../../shared/Table';
import ListItem from './ListItem';
import { colors, borderRadius } from '../../../style/variables';
import { get } from 'lodash';
import { TRANSACTION_ID } from '../../../../common/constants';

const TransactionsContainer = styled.div`
  position: relative;
  overflow: hidden;
  padding: 0;
  border: 1px solid ${colors.elementBorder};
  border-radius: ${borderRadius};
`;

function TransactionList({ appName, list, type }) {
  const transactions = list.data || [];

  return (
    <TransactionsContainer>
      <Table>
        <thead>
          <tr>
            <TableHead>Endpoints</TableHead>
            <TableHead>Avg. resp. time</TableHead>
            <TableHead>95th percentile</TableHead>
            <TableHead>RPM</TableHead>
            <TableHead>Impact</TableHead>
          </tr>
        </thead>

        <tbody>
          <TableLoader status={list.status} columns={5} />

          {transactions.map(transaction => {
            return (
              <ListItem
                key={get({ transaction }, TRANSACTION_ID)}
                appName={appName}
                type={type}
                transaction={transaction}
              />
            );
          })}
        </tbody>
      </Table>
    </TransactionsContainer>
  );
}

export default TransactionList;
