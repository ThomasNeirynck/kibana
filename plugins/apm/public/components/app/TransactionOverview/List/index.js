import React from 'react';
import styled from 'styled-components';
import { Table, TableHead, TableLoader } from '../../../shared/Table';
import ListItem from './ListItem';

import { units, px, colors, borderRadius } from '../../../../style/variables';
import { get } from 'lodash';
import { TRANSACTION_ID } from '../../../../../common/constants';
import { KuiInfoButton } from 'ui_framework/components';
import { Tooltip } from 'pui-react-tooltip';
import { OverlayTrigger } from 'pui-react-overlay-trigger';

const TransactionsContainer = styled.div`
  position: relative;
  overflow: hidden;
  padding: 0;
  border: 1px solid ${colors.elementBorder};
  border-radius: ${borderRadius};
`;

const Wrapper = styled.div`
  position: relative;
  display: inline-block;
  top: 1px;
  left: ${px(units.half)};
`;

const ImpactToolTip = () => (
  <Wrapper>
    <OverlayTrigger
      placement="top"
      trigger="click"
      overlay={
        <Tooltip>
          Impact shows the most used and slowest endpoints in your app.
        </Tooltip>
      }
    >
      <KuiInfoButton />
    </OverlayTrigger>
  </Wrapper>
);

const getRelativeImpact = (impact, impactMin, impactMax) =>
  Math.max((impact - impactMin) / Math.max(impactMax - impactMin, 1) * 100, 1);

function TransactionList({
  appName,
  list,
  type,
  changeTransactionSorting,
  transactionSorting
}) {
  const transactions = list.data;

  const impacts = transactions.map(({ impact }) => impact);
  const impactMin = Math.min(...impacts);
  const impactMax = Math.max(...impacts);

  return (
    <TransactionsContainer>
      <Table>
        <thead>
          <tr>
            {[
              { key: 'name', label: 'Name' },
              { key: 'avg', label: 'Avg. resp. time' },
              { key: 'p95', label: '95th percentile' },
              { key: 'rpm', label: 'RPM' }
            ].map(({ key, label }) => (
              <TableHead
                key={key}
                onClick={() => changeTransactionSorting(key)}
                selected={transactionSorting.key === key}
                descending={transactionSorting.descending}
              >
                {label}
              </TableHead>
            ))}

            <TableHead
              onClick={() => changeTransactionSorting('impact')}
              selected={transactionSorting.key === 'impact'}
              descending={transactionSorting.descending}
            >
              Impact
              <ImpactToolTip />
            </TableHead>
          </tr>
        </thead>

        <tbody>
          <TableLoader status={list.status} columns={5} />

          {transactions.map(transaction => {
            return (
              <ListItem
                key={get({ transaction }, TRANSACTION_ID)}
                appName={appName}
                type={type}
                transaction={transaction}
                impact={getRelativeImpact(
                  transaction.impact,
                  impactMin,
                  impactMax
                )}
              />
            );
          })}
        </tbody>
      </Table>
    </TransactionsContainer>
  );
}

export default TransactionList;
