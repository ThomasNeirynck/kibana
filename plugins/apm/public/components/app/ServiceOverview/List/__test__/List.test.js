import React from 'react';
import { mount } from 'enzyme';
import toDiffableHtml from 'diffable-html';
import { MemoryRouter } from 'react-router-dom';
import List from '../index';
import props from './props.json';
import {
  mountWithRouterAndStore,
  mockMoment
} from '../../../../../utils/testHelpers';

describe('ErrorGroupOverview -> List', () => {
  beforeAll(() => {
    mockMoment();
  });

  it('should render empty state', () => {
    const storeState = {};
    const wrapper = mount(
      <MemoryRouter>
        <List items={[]} />
      </MemoryRouter>,
      storeState
    );

    expect(toDiffableHtml(wrapper.html())).toMatchSnapshot();
  });

  it('should render with data', () => {
    const storeState = { location: {} };
    const wrapper = mountWithRouterAndStore(
      <List
        items={props.items}
        changeServiceSorting={props.changeServiceSorting}
        serviceSorting={props.serviceSorting}
      />,
      storeState
    );

    expect(toDiffableHtml(wrapper.html())).toMatchSnapshot();
  });
});
