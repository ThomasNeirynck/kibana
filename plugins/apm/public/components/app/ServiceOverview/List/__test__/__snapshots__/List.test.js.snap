// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorGroupOverview -> List should render empty state 1`] = `
"
<div class=\\"kuiControlledTable\\">
  <div class=\\"kuiToolBar\\">
    <div class=\\"kuiToolBarSearch\\">
      <div class=\\"kuiToolBarSearchBox\\">
        <div class=\\"kuiToolBarSearchBox__icon kuiIcon fa-search\\">
        </div>
        <input type=\\"text\\"
               class=\\"kuiToolBarSearchBox__input\\"
               placeholder=\\"Filter...\\"
               aria-label=\\"Filter\\"
        >
      </div>
    </div>
    <div class=\\"kuiToolBarSection\\">
      <div class=\\"kuiPager\\">
        <div class=\\"kuiPagerText\\">
          0–0 of 0
        </div>
        <div class=\\"kuiButtonGroup kuiButtonGroup--united\\"
             role=\\"group\\"
        >
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show previous page\\"
                  data-test-subj=\\"pagerPreviousButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-left\\"
              >
              </span>
            </span>
          </button>
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show next page\\"
                  data-test-subj=\\"pagerNextButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-right\\"
              >
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div class=\\"kuiPanel kuiPanel--centered kuiPanel--withToolBar\\">
    <div class=\\"kuiTableInfo sc-bdVaJa cotoTP\\">
      No services with data in the selected time range.
      <div class=\\"sc-bwzfXH ePpHnj\\">
        <span>
          Oops! You should try another time range. If that's no good, there's always the
          <a href=\\"https://www.elastic.co/guide/en/apm/get-started/6.2/index.html\\">
            documentation
          </a>
          .
        </span>
      </div>
    </div>
  </div>
  <div class=\\"kuiToolBarFooter\\">
    <div class=\\"kuiToolBarFooterSection\\">
      <div class=\\"sc-htpNat emgphq\\">
      </div>
    </div>
    <div class=\\"kuiToolBarSection\\">
      <div class=\\"kuiPager\\">
        <div class=\\"kuiPagerText\\">
          0–0 of 0
        </div>
        <div class=\\"kuiButtonGroup kuiButtonGroup--united\\"
             role=\\"group\\"
        >
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show previous page\\"
                  data-test-subj=\\"pagerPreviousButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-left\\"
              >
              </span>
            </span>
          </button>
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show next page\\"
                  data-test-subj=\\"pagerNextButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-right\\"
              >
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
"
`;

exports[`ErrorGroupOverview -> List should render with data 1`] = `
"
<div class=\\"kuiControlledTable\\">
  <div class=\\"kuiToolBar\\">
    <div class=\\"kuiToolBarSearch\\">
      <div class=\\"kuiToolBarSearchBox\\">
        <div class=\\"kuiToolBarSearchBox__icon kuiIcon fa-search\\">
        </div>
        <input type=\\"text\\"
               class=\\"kuiToolBarSearchBox__input\\"
               placeholder=\\"Filter...\\"
               aria-label=\\"Filter\\"
        >
      </div>
    </div>
    <div class=\\"kuiToolBarSection\\">
      <div class=\\"kuiPager\\">
        <div class=\\"kuiPagerText\\">
          0–2 of 2
        </div>
        <div class=\\"kuiButtonGroup kuiButtonGroup--united\\"
             role=\\"group\\"
        >
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show previous page\\"
                  data-test-subj=\\"pagerPreviousButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-left\\"
              >
              </span>
            </span>
          </button>
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show next page\\"
                  data-test-subj=\\"pagerNextButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-right\\"
              >
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <table class=\\"kuiTable\\">
    <thead>
      <tr>
        <th class=\\"kuiTableHeaderCell sc-bxivhb cgZttI\\"
            scope=\\"col\\"
        >
          <button class=\\"kuiTableHeaderCellButton kuiTableHeaderCellButton-isSorted\\"
                  aria-label=\\"Sort Name descending\\"
          >
            <span class=\\"kuiTableHeaderCell__liner\\">
              Name
              <span class=\\"kuiTableSortIcon kuiIcon fa-long-arrow-up\\"
                    aria-hidden=\\"true\\"
              >
              </span>
            </span>
          </button>
        </th>
        <th class=\\"kuiTableHeaderCell sc-bxivhb cgZttI\\"
            scope=\\"col\\"
        >
          <button class=\\"kuiTableHeaderCellButton\\"
                  aria-label=\\"Sort Agent descending\\"
          >
            <span class=\\"kuiTableHeaderCell__liner\\">
              Agent
              <span class=\\"kuiTableSortIcon kuiIcon fa-long-arrow-up\\"
                    aria-hidden=\\"true\\"
              >
              </span>
            </span>
          </button>
        </th>
        <th class=\\"kuiTableHeaderCell kuiTableHeaderCell--alignRight sc-bxivhb cgZttI\\"
            scope=\\"col\\"
        >
          <button class=\\"kuiTableHeaderCellButton\\"
                  aria-label=\\"Sort Avg. response time descending\\"
          >
            <span class=\\"kuiTableHeaderCell__liner\\">
              Avg. response time
              <span class=\\"kuiTableSortIcon kuiIcon fa-long-arrow-up\\"
                    aria-hidden=\\"true\\"
              >
              </span>
            </span>
          </button>
        </th>
        <th class=\\"kuiTableHeaderCell kuiTableHeaderCell--alignRight sc-bxivhb cgZttI\\"
            scope=\\"col\\"
        >
          <button class=\\"kuiTableHeaderCellButton\\"
                  aria-label=\\"Sort Trans. per minute descending\\"
          >
            <span class=\\"kuiTableHeaderCell__liner\\">
              Trans. per minute
              <span class=\\"kuiTableSortIcon kuiIcon fa-long-arrow-up\\"
                    aria-hidden=\\"true\\"
              >
              </span>
            </span>
          </button>
        </th>
        <th class=\\"kuiTableHeaderCell kuiTableHeaderCell--alignRight sc-bxivhb cgZttI\\"
            scope=\\"col\\"
        >
          <button class=\\"kuiTableHeaderCellButton\\"
                  aria-label=\\"Sort Errors per minute descending\\"
          >
            <span class=\\"kuiTableHeaderCell__liner\\">
              Errors per minute
              <span class=\\"kuiTableSortIcon kuiIcon fa-long-arrow-up\\"
                    aria-hidden=\\"true\\"
              >
              </span>
            </span>
          </button>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"kuiTableRow\\">
        <td class=\\"kuiTableRowCell sc-EHOje jXLSDR\\">
          <div class=\\"kuiTableRowCell__liner\\">
            <a class=\\"kuiLink sc-gzVnrw hfGdFM\\"
               aria-describedby=\\"overlay1\\"
               href=\\"#/opbeans-node/transactions\\"
            >
              opbeans-node
            </a>
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ\\">
          <div class=\\"kuiTableRowCell__liner\\">
            nodejs
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            N/A
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            0 tpm
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            46.1 err.
          </div>
        </td>
      </tr>
      <tr class=\\"kuiTableRow\\">
        <td class=\\"kuiTableRowCell sc-EHOje jXLSDR\\">
          <div class=\\"kuiTableRowCell__liner\\">
            <a class=\\"kuiLink sc-gzVnrw hfGdFM\\"
               aria-describedby=\\"overlay2\\"
               href=\\"#/opbeans-python/transactions\\"
            >
              opbeans-python
            </a>
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ\\">
          <div class=\\"kuiTableRowCell__liner\\">
            python
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            92 ms
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            86.9 tpm
          </div>
        </td>
        <td class=\\"kuiTableRowCell sc-bZQynM ckLYrQ kuiTableRowCell--alignRight\\">
          <div class=\\"kuiTableRowCell__liner\\">
            12.6 err.
          </div>
        </td>
      </tr>
    </tbody>
  </table>
  <div class=\\"kuiToolBarFooter\\">
    <div class=\\"kuiToolBarFooterSection\\">
      <div class=\\"sc-htpNat emgphq\\">
      </div>
    </div>
    <div class=\\"kuiToolBarSection\\">
      <div class=\\"kuiPager\\">
        <div class=\\"kuiPagerText\\">
          0–2 of 2
        </div>
        <div class=\\"kuiButtonGroup kuiButtonGroup--united\\"
             role=\\"group\\"
        >
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show previous page\\"
                  data-test-subj=\\"pagerPreviousButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-left\\"
              >
              </span>
            </span>
          </button>
          <button class=\\"kuiButton kuiButton--basic kuiButton--iconText\\"
                  aria-label=\\"Show next page\\"
                  data-test-subj=\\"pagerNextButton\\"
                  disabled
          >
            <span class=\\"kuiButton__inner\\">
              <span aria-hidden=\\"true\\"
                    class=\\"kuiButton__icon kuiIcon fa-chevron-right\\"
              >
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
"
`;
